pkg/controller/jobs/job/job_webhook.go

```diff
@@ -117,6 +124,15 @@ func validatePartialAdmissionCreate(job *Job) field.ErrorList {
 			}
 		}
 	}
+	if strVal, found := job.Annotations[JobSyncCompletionWithParallelism]; found {
+		enabled, err := strconv.ParseBool(strVal)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(syncCompletionAnnotationsPath, job.Annotations[JobSyncCompletionWithParallelism], err.Error()))
+		}
+		if enabled && pointer.Int32Deref(job.Spec.Parallelism, 1) != pointer.Int32Deref(job.Spec.Completions, 1) {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "completions"), job.Spec.Completions, "should be equal to parallelism when sync completions is enabled"))
+		}
+	}
```

Comment:
Should we add a validation to reject jobs with `NonIndexed` mode and `kueue.x-k8s.io/job-completions-equal-parallelism=true` annotation?
IIUC, the `completions` field is immutable when the job is created with `NonIndexed` mode.

Reply:
+1

Reply:
done